### DISCUSSION TITLE
Custom EPS32-h2 device

### DISCUSSION THREAD

**Discussion by Omnia6975** [Koenkk/zigbee2mqtt#19339] [Devices]
Hi,

I have a custom end Zigbee device I made with an ESP32-H2 board.
I want to report some events which occurred from sensors.
From a Zigbee sniffer I got :

![image](https://github.com/Koenkk/zigbee2mqtt/assets/81617292/12733234-a31f-421f-9610-bbd2c1d90188)

On zigbee2mqtt when I send this "report" packet I got this :

` MQTT publish: topic 'zigbee2mqtt/0x4831b7fffec070ba', payload '{"Stopped event":null,"linkquality":112,"state":1}'`

My custom converter is defined as below :

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const fzLocal = {
    Oia_blind: {
        cluster: 'genBinaryInput',
        type: 'attributeReport',
        convert: (model, msg, publish, options, meta) => {
            return {state: msg.data.presentValue};
        }
    },
};

const definition = {
    zigbeeModel: ['BLIND_2'], 
    model: 'BLIND_2', 
    vendor: 'Oia',
    description: 'Custom blind module',
    fromZigbee: [fzLocal.Oia_blind],
    toZigbee: [tz.on_off],
    exposes: [
        e.switch().withState('Stop action', true),
        exposes.binary('Stopped event', ea.STATE, true, false)
        ],
    configure: async (device, coordinatorEndpoint, logger) => {
    },
};

module.exports = definition;
```

I can even see the state changes in the state tab of the zigbee device on HA
![image](https://github.com/Koenkk/zigbee2mqtt/assets/81617292/2846354b-2177-4479-b571-161886daae29)

My issue is that I am not able to use this 'report' event in the list of entities to do some automations. Indeed, this property is not visible in the History Graph Card in the entities fields available.

My 4 questions : 

- [ ] A/  I am pretty sure it is due to the wrong definition of my converter. Could you please help me to fix it  ?

- [ ] B/ I have also a switch defined in the converter. Is it possible to define a push button in the converter instead of a switch ? (In my application the state information provided does not make sense, it is used to launch an action instead of a state change. If it is possible could you please help me to properly define the converter please ?
- [ ] C/ I have not been able to change the state  of this switch from the end device. For instance and contrary to what I have done at that time with one button and one report property,  is it possible with only one switch defined in the converter, to be able to change the sate from the 'Expose' tab of HA (A send command from HA which will change the visual state of the switch) but also be able to update this state from the device itself to the 'Expose' tab of HA  (send command from the end device to HA). If it is possible, could you please help me to properly define the converter please ?
- [ ] D/ More generally, I have some difficulties to understand how the zigbee-herdsman-converters is working and I did not fin a good documentation about it. At that time I use the devices implementation and browse the source code. Do you have any useful links or advices for that purpose ?

**Comment by Omnia6975** [comment]
Any help or advice, please ? Should I post it to another thread ?
I am still stuck with my issue.
Thanks in advance

**Comment by Omnia6975** [comment]
Up please

**Comment by mlouaze** [comment]
Hi, I know this thread is old but did you find a solution?

**Reply by Omnia6975** [reply]
I did not find any answer to theses questions so far. :(
What's about you ?

**Comment by zhechenzhu** [comment]
I saw much more active dicussion on the topic of supporting the new esp32 zigbee boards at esphome. There is even a working example there. Haven't dived deep enough but looks promising.
